core/domain/exp_domain.py

```diff
@@ -1179,10 +1183,11 @@ def validate(self, exp_param_specs_dict):
             raise utils.ValidationError(
                 'Terminal interactions must not have any answer groups.')
 
-        for answer_group in self.answer_groups:
-            answer_group.validate(interaction, exp_param_specs_dict)
-        if self.default_outcome is not None:
-            self.default_outcome.validate()
+        if not is_question:
+            for answer_group in self.answer_groups:
+                answer_group.validate(interaction, exp_param_specs_dict)
```

Comment:
What about validating other aspects of ``AnswerGroup``? Is there anything in the outcome that's worth validating in both cases? Should the ``is_question`` check be delegated to ``Outcome`` to decide how to validate?

Reply:
All of the checks related to answer groups and outcomes exclusive to Question, I am doing in question.validate() itself.